trace.go

```diff
@@ -344,7 +344,7 @@ func (t *pubsubTracer) traceRPCMeta(rpc *RPC) *pb.TraceEvent_RPCMeta {
 	var msgs []*pb.TraceEvent_MessageMeta
 	for _, m := range rpc.Publish {
 		msgs = append(msgs, &pb.TraceEvent_MessageMeta{
-			MessageID: []byte(t.msgID(m)),
+			MessageID: []byte(t.idGen.ID(&Message{Message: m})),
```

Comment:
No way to get `Message`, but should not be a big issue.

Reply:
I think we should make a method in idGen that does this, no need to alloc here.

Reply:
Or we can expose a secondary method, refactor out of ID (so that it calls it) and call it RawID or something?
It will be nicer, so we might as well do it.

Reply:
Good idea. Done